### ISSUE TITLE
repr for Tensor

### ISSUE THREAD

**Issue by mganahl** [google/TensorNetwork#814]
We need to add a useful `__repr__` to `Tensor`

**Comment by gaxler** [comment]
Shape and type seems like a useful information.
Should we have a string representation of an array as part of Tensor repr?

It seems to me that Tensor class is used with linalg functions, in this case it might be annoying to call Tensor.array to examine array's content.  What do you think?

**Comment by mganahl** [comment]
I think the repr of the underlying array is fine.

gaxler <notifications@github.com> schrieb am Di. 8. Sep. 2020 um 22:18:

**Comment by Thenerdstation** [comment]
Agreed. I think we'd be ok to to fall back on however the underlying array does it's `__repr__` + some extra info?

**Comment by gaxler** [comment]
If we gonna use underlying repr we might get a lot of duplicate info in the repr or inconsistent repr depending on the backend.

e.g. TensorFlow has shape and dtype, pytorch and numpy might have dtype
> <tf.Tensor: shape=(10, 10, 10), dtype=float32, numpy=...>
> array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1], dtype=int16)

So maybe just shape + type and leave the rest for Tensor.array?

**Comment by wkostuch** [comment]
Hey, does this still need to be done?  I'm interested in working on it but this would be my first open-source contribution which is a little daunting.

**Comment by mganahl** [comment]
Hi! This is addressed in PR #823.  You can add @alewis' comments to #823 if you want!

**Comment by alewis** [comment]
> Hey, does this still need to be done? I'm interested in working on it but this would be my first open-source contribution which is a little daunting.

I think we're good for this one (although indeed you could make the small change suggested in #823 as a first step). However, we have many things that would be a great way to dip your toes into contributing. #840 in particular would have you moving around through most of the code, making small changes, and writing tests - having done it you should be pretty much equipped to do anything. I'd be thrilled to walk you through it if you're interested.

**Comment by wkostuch** [comment]
@alewis I'll go pop over to #840, thanks for the suggestion!
